pkg/eraser/eraser_test.go

```diff
@@ -43,13 +44,22 @@ func TestParseEndpointWithFallBackProtocol(t *testing.T) {
 			fallbackProtocol: "unix",
 			protocol:         "",
 			addr:             "",
-			err:              ErrParsing,
+			err:              &url.Error{},
```

Comment:
WDYT if instead oof a literal error we do something like passing a `func(t *testing.T, err error)` to do the error check.
So in the test cases the `err` field would be a closure, and in the actual test code below we pass it like `tc.errCheck(t, err)`.

I think that gets rid of this weird case with url.Error, and we can be more specific with the check.